OpenRA.Mods.Common/Traits/Upgrades/DisableOnUpgrade.cs

```diff
@@ -18,12 +18,13 @@ public class DisableOnUpgradeInfo : UpgradableTraitInfo
 		public override object Create(ActorInitializer init) { return new DisableOnUpgrade(this); }
 	}
 
-	public class DisableOnUpgrade : UpgradableTrait<DisableOnUpgradeInfo>, IDisable, IDisableMove
+	public class DisableOnUpgrade : UpgradableTrait<DisableOnUpgradeInfo>, IDisable, IDisableMove, IPowerModifier
 	{
 		public DisableOnUpgrade(DisableOnUpgradeInfo info)
 			: base(info) { }
 
 		public bool Disabled { get { return !IsTraitDisabled; } }
 		public bool MoveDisabled(Actor self) { return !IsTraitDisabled; }
+		public int GetPowerModifier() { return IsTraitDisabled ? 100 : 0; }
```

Comment:
`PowerMultiplier` can be used to do the same thing:
```yaml
	PowerMultiplier@EMPDISABLE:
		UpgradeTypes: empdisable
		Modifier: 0
```

Other effects (Iron Curtain) use this pattern.